### ISSUE TITLE
Integration With Anyhow

### ISSUE THREAD

**Issue by amab8901** [tokio-rs/tracing#2985]
### Motivation

Reduce noise in the code

### Proposal

The code would look cleaner if I could write like this:
``` rust
use std::str::FromStr;

use anyhow::Context;
use num::BigUint;
use num::ToPrimitive;
use tracing::error;

fn main() {
    let uint = BigUint::from_str("1").unwrap();
    let floating_uint = uint
        .to_f64()
        .error("Failed to convert `BigUint` to `f64`.") // diff
        .ok()
        .unwrap();
}

```

Doesn't have to be called `error`. Some other suggestions are `tracing_error`, `t_error`, `tracing_err`. And same thing for the other log levels (info, warn, trace, etc)

### Alternatives

Here is how I'm currently writing the code:
``` rust
use std::str::FromStr;

use anyhow::Context;
use num::BigUint;
use num::ToPrimitive;
use tracing::error;

fn main() {
    let uint = BigUint::from_str("1").unwrap();
    let floating_uint = uint
        .to_f64()
        .context("Failed to convert `BigUint` to `f64`.") // diff
        .map_err(|err| error!("{err}")) // diff
        .ok()
        .unwrap();
}

```

**Comment by mladedav** [comment]
I don't think this is something that should really be added. The second variant seems good to me and it actually allows you to do more things with the error, e.g. `.map_err(|err| tracing::error!(error.debug = ?debug, error.display = %error, more_stuff, "Additional context"))`. I feel like creating an event with the error message alone is often not the thing that I'd want to do.

As an aside, you can also use `inspect_err` instead since you're not consuming the error, your code as a side-effect changes the error type to `()`.

Nevertheless, it should be pretty easy for any crate to create an extension trait for `Result` similarly to what `anyhow::Context` does and the `tracing::event!` there if you want this functionality.
